### ISSUE TITLE
MP REST Client should implement a type of connection pooling

### ISSUE THREAD

**Issue by bpaskin** [eclipse/microprofile-rest-client#242]
Currently Java allows for a number of maximum connections to destinations based on the property `http.maxConnections` and can be kept alive using `http.keepAlive`.  However, this may not be fined grained enough for applications that need to communicate with multiple systems that will have varying degrees of connections allowed.

It would be a good feature to have some sort of connection pooling for each destination, similar to that of datasources, that allow for:

- number of connections
- keep-alive timeout
- maximum request time before throwing an error and disconnecting
- validation of connections
etc

-Brian

**Comment by kenfinnigan** [comment]
I believe most implementations of MP REST Client have connection pooling in their underlying HTTP Client, but we could investigate adding properties to "standardize" how to it's configured.

For maximum request time, that's handled with the existing `connectTimeout` and `readTimeout` properties, along with combining it with MP Fault Tolerance annotations

**Comment by bpaskin** [comment]
Agreed with the `connectionTimeout` and `readTimeout`, but unfortunately not al devs follow good programming practices.

**Comment by Kiiv** [comment]
Really interested in this feature too. I'm using Open Liberty and I don't find any solution to access the underlying HTTP Client and set max connections :s

**Comment by tushar-1728** [comment]
So, do we any update on the implementation of this feature?

**Comment by Emily-Jiang** [comment]
@njr-11 any thoughts on this?

**Comment by njr-11** [comment]
> @njr-11 @andymc12 any thoughts on this?

For REST Client on Liberty, the correct contact is @jim-krueger

**Comment by jim-krueger** [comment]
Connection pooling for Open Liberty today is a per destination size.   Standardizing configuration of this is something that maybe should be considered, but I'm not sure if this belongs with microprofile-rest-client.   I think it might be a better candidate for RESTful Web Services.
